### PR TITLE
Updates golang-dind go version to 1.13.4

### PR DESCRIPTION
**Pull request by JoshVanL**
Signed-off-by: JoshVanL <vleeuwenjoshua@gmail.com>

### CODE REVIEW

File: images/golang-dind/build.yaml [hunk 1]
```diff
@@ -4,3 +4,8 @@ name: golang-dind # Name of the image to be built
 images:
 - ${_REGISTRY}/${_NAME}:${_DATE_STAMP}-${_GIT_REF}
 - ${_REGISTRY}/${_NAME}:latest
```
Comment (munnerz) [inline images/golang-dind/build.yaml]: You should update the `images` list here to include `GO_VERSION`, e.g.:

```
# Image names to be tagged and pushed
images:
- ${_REGISTRY}/${_NAME}:${_DATE_STAMP}-${_GIT_REF}-${BAZEL_VERSION}
- ${_REGISTRY}/${_NAME}:latest-${BAZEL_VERSION}
```

from: https://github.com/jetstack/testing/blob/master/images/bazelbuild/build.yaml

File: images/golang-dind/build.yaml [hunk 2]
```diff
@@ -1,6 +1,12 @@
 name: golang-dind # Name of the image to be built
 
+variants:
+  "1.13.4":
+    arguments:
+      BASE_IMAGE: "${_REGISTRY}/bazelbuild:latest-1.0.0"
```
Comment (munnerz) [inline images/golang-dind/build.yaml]: Probably best to pin to a specific version here, else it'll be difficult to know exactly what base image it ended out being built with.

File: images/golang-dind/Dockerfile [hunk 3]
```diff
@@ -92,3 +58,9 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     patch \
     && apt-get clean \
     && python -m pip install --upgrade pip setuptools wheel
```
Comment (munnerz) [inline images/golang-dind/Dockerfile]: These lines and above are already performed in the bazelbuild image build: https://github.com/jetstack/testing/blob/6b2a9745faf5a307113b94aa73580fde720ac8e8/images/bazelbuild/Dockerfile#L66-L92

File: images/golang-dind/build.yaml (hunk 2 above)
Comment (munnerz) [reply images/golang-dind/build.yaml]: e.g. eu.gcr.io/jetstack-build-infra-images/bazelbuild:20191016-eff358a-1.0.0

File: images/golang-dind/build.yaml [hunk 4]
```diff
@@ -1,6 +1,12 @@
 name: golang-dind # Name of the image to be built
 
+variants:
+  "1.13.4":
+    arguments:
+      BASE_IMAGE: "${_REGISTRY}/bazelbuild:20191016-eff358a-1.0.0"
```
Comment (munnerz) [inline images/golang-dind/build.yaml]: Not sure if this will work but let’s find out!

/lgtm
/approve

**Comment by JoshVanL** [comment]
@munnerz did this work? I'm not seeing the image in gcr

**Pull request merged**